internal/pkg/helm/convert.go

```diff
@@ -375,3 +378,73 @@ func InstallAnsibleTasks(roleDirectory string) {
 			destinationTasksYamlFile)
 	}
 }
+
+<<<<<<< HEAD
+// Determines whether a YAML variable (i.e., "image.pullPolicy") refers to a boolean within a given Helm Chart context.
+// TODO: This is currently unused, as the testing method skips client use.  In the future, call this from cmd.go.
+func IsBooleanYamlsValue(chartClient HelmChartClient, variableName string) (bool, error){
+	variableArray := strings.Split(variableName, goTemplateMemberAccessOperator)
+
+	raw, err := chartutil.ReadValues([]byte(chartClient.Chart.Values.Raw))
+	if err != nil {
+		return false, errors.New("Could not parse " + variableName)
+	}
+
+	// Determine depth and make sure that the path that was passed has at least one element.
+	variableDepth := len(variableArray)
+	if variableDepth < 1 {
+		return false, errors.New("Cannot split " + variableName + " appropriately")
+	}
+
+	return IsBooleanYamlValue(raw, variableArray)
+}
+
+// Determines whether path within the input context refers to a boolean type.  Consulting a Helm Chart's values is
+// helpful for determining whether a Helm Chart template conditional is checking for boolean equality v.s. definition.
+func IsBooleanYamlValue(input map[string]interface{}, path []string) (bool, error) {
+	if input == nil {
+		return false, errors.New("input cannot be nil")
+	}
+	if path == nil || len(path) < 1 {
+		return false, errors.New("path slice must have at least one element")
+	}
+	return isBooleanYamlValue(input, path, 0)
+}
+
+// Internal recursive helper function that determines whether path within the input context refers to a boolean type.
+func isBooleanYamlValue(input map[string]interface{}, path []string, i int) (bool, error) {
+	pathKey := path[i]
+
+	// Recursive base case.  When we have reached the last sub-path within a path (i.e., "pullPolicy" for
+	// path=["metrics", "image", "pullPolicy"]
+	if i == len(path) - 1 {
```

Comment:
use constant for magic numbers

Reply:
It isn't really a magic number;  it is a stopping path index count.  I moved the logic to the exported helper function "IsBooleanYamlValue, and documented the stopping condition a little more verbosely.